### DISCUSSION TITLE
Reward feature for habits

### DISCUSSION THREAD

**Discussion by shivanshsharma13** [iSoron/uhabits#1521] [Feature Requests]
Hi, community. First of all great job building this mobile app it's one of the best habit tracking apps I would say.

I was thinking if we could take this app with the reward feature. Like in books like *Atomic Habits* and *The power of habit* The habit loop always contains some reward that the user selects in their own way.

```mermaid
stateDiagram
    Cue --> Craving
		Craving --> Responce
    Responce --> reward
		reward --> Cue
		
		cue --> Routine
		Routine --> Reward
    Reward --> cue
```

## Feature discription -

We do have

- **Cue** - Time of doing an activity
- **Craving & response** -  Make it attractive and motivated (we do have good progress charts and streaks for that)
- **Reward**

(We kind of have a reward, like streaks and progress reports) but I thought what if the user has the ability to enter a reward for their habit and set their goals like-

|Habit| Reward tracking| reward|
|-|-|-|
|Meditation| Regular Meditation for 30 days for at least 10 min| If I would do so I'll get Noice canceling headphones for myself as a reward|
|Gym| Regular gym for 30 days| If I would do so I'll get a game pass for myself as a reward.|
|Running| Running for 30 days| Sport shoes| 

## Feature use case

The use cases are-

-  As of now we have a notification feature like *Did you go to the gym?* and user type their own. How about having a notification like **20 Days left for the reward. Stay motivated 🙂**
-  We can add daily, weekly or monthly rewards. The user has an option to do so. 

## Feature Implementation

Coming to the implementation we can add a reward option on the creation of the habit page. 

- The reward option consists of 2 fields

    - Reward name
    - Way of tracking (daily, weekly, or monthly)

- Notifications
	- *Daily* has notifications like **Your reward_name is waiting for you, stay committed**
	- *Weekly* has normal notifications like **You have 7 days left for the reward_name Stay motivated 🙂**
	- *Monthly* has a notification like **20 Days left for the reward. Stay motivated 🙂**
- The reward will be tracked from the **Target** chart we have in **measurement type** and total days on **Non-measurement types**  habits.
- On successful completion of habit we can have a bottom slider message for **Congratulations, You can earn your reward.**

What are your views on it? We are open to other ideas for the same.

**Comment by t214c** [comment]
Can this get looked at? There are no open-source habit/task-rewarding apps so that should give it some priority.

**Reply by RetriedAgain** [reply]
Habitica is an open-source habit/task-rewarding app with a freemium model. See their rewards-system here.
https://habitica.fandom.com/wiki/Rewards

**Comment by RetriedAgain** [comment]
This could possibly interfere with the current minimalistic structure of the app, which is advertised in the README.

> Beautiful, minimalistic and lightweight interface. Loop has an elegant and minimalistic interface that is very easy to use, even for first-time users. 

I do not speak for the project, of course.
